### PR TITLE
allow C-style casts in c code

### PR DESCRIPTION
**Pull request by dirk-thomas**

### CODE REVIEW

File: ament_cpplint/ament_cpplint/cpplint.py [hunk 1]
```diff
@@ -5432,6 +5432,10 @@ def CheckCStyleCast(filename, clean_lines, linenum, cast_type, pattern, error):
           'All parameters should be named in a function')
     return True
 
+  # Don't warn in C files about C-style casts
+  if os.path.splitext(filename)[1] in ['.c', '.cc', '.h', '.hh']:
```
Comment (wjwwood) [inline ament_cpplint/ament_cpplint/cpplint.py]: `.cc` and `.hh` are for C++ aren't they?

Comment (dirk-thomas) [inline ament_cpplint/ament_cpplint/cpplint.py]: Fixed.

**Comment by esteve** [comment]
+1

**Comment by wjwwood** [comment]
+1

**Pull request merged**